src/components/actions/executeAction.ts

```diff
@@ -0,0 +1,98 @@
+import { ActionLibrary, ActionPayload, ActionQueueExecutePayload } from '../../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+import functions from './evaluatorFunctions'
+import DBConnect from '../databaseConnect'
+import fetch from 'node-fetch'
+import { EvaluatorNode } from '@openmsupply/expression-evaluator/lib/types'
+import { getApplicationData } from './getApplicationData'
+import { ActionQueueStatus } from '../../generated/graphql'
+import config from '../../config'
+import { evaluateParameters } from './helpers'
+
+// Dev config
+const showApplicationDataLog = false
+
+const graphQLEndpoint = config.graphQLendpoint
+
+export async function executeAction(
+  payload: ActionPayload,
+  actionLibrary: ActionLibrary,
+  additionalObjects: any = {}
+): Promise<ActionQueueExecutePayload> {
+  // Get fresh applicationData for each Action
+  const applicationData = await getApplicationData({ payload })
+
+  // Debug helper console.log to inspect applicationData:
+  if (showApplicationDataLog) console.log('ApplicationData: ', applicationData)
+
+  const evaluatorParams = {
+    objects: { applicationData, functions, ...additionalObjects },
+    pgConnection: DBConnect,
+    APIfetch: fetch,
+    graphQLConnection: { fetch, endpoint: graphQLEndpoint },
+  }
+
+  // Evaluate condition
+  let condition
+  try {
+    condition = await evaluateExpression(
+      payload.condition_expression as EvaluatorNode,
+      evaluatorParams
+    )
+  } catch (err) {
+    console.log('>> Error evaluating condition for action:', payload.code)
+    return await DBConnect.executedActionStatusUpdate({
+      status: ActionQueueStatus.Fail,
+      error_log: 'Problem evaluating condition: ' + err,
+      parameters_evaluated: null,
+      output: null,
+      id: payload.id,
+    })
+  }
+
+  if (condition) {
```

Comment:
could do early return here if !condition

Reply:
Sure: b8d192c0

Haven't tested, but seems like it should be identical behaviour.